modules/restore-determine-state-etcd-member.adoc

```diff
@@ -0,0 +1,78 @@
+
+The steps to replace an unhealthy etcd member depend on which of the following states your etcd member is in:
+
+* The node is not running.
+* The node is running, but etcd is crashlooping.
+
+The procedure determines which state your etcd member is in, so that you know which procedure to follow to replace the unhealthy etcd member.
+
+.Prerequisites
+
+* You have access to the cluster as a user with the cluster-admin role.
+* You have identified an unhealthy etcd member.
+
+.Procedure
+
+. Determine if the *node is not running*.
++
+If any of the following scenarios are true, then the *node is not running*.
+
+** Check if the machine for the unhealthy member is not running.
++
+----
+$ oc get machines -A -ojsonpath='{range ..items[?(@.status.nodeRef.name=="ip-10-0-131-183.ec2.internal")]}{@.status.providerStatus.instanceState}{"\n"}'
+stopped <1>
+----
+<1> If this output is anything other than `running`, then the *node is not running*.
++
+// TODO: Need to change this command so that it's easier for the user to replace the node with theirs
+
+** If the machine is running, then check whether the node is unreachable.
++
+----
+$ oc get nodes -o json | jq '.items[] | select(.spec.taints[]?.key=="node.kubernetes.io/unreachable") | .metadata.name' | uniq
+"Ip-10-0-131-183.ec2.internal" <1>
+----
+<1> If the node is listed, then the *node is not running*.
++
+// TODO: Change this command to not use jq
+
+** If the node is still reachable, check whether the node is listed as `NotReady`:
++
+----
+$ oc get nodes -l node-role.kubernetes.io/master | grep "NotReady"
+ip-10-0-131-183.ec2.internal   NotReady   master   122m   v1.17.1 <1>
+----
+<1> If the node is listed as `NotReady`, then the *node is not running*.
+
+. Determine if the *node is running, but etcd is crashlooping*.
++
+If the previous steps determined that the node is still running, follow these steps to determine if etcd is crashlooping.
+
+.. Verify that all master nodes are listed as `Ready`:
++
+----
+$ oc get nodes -l node-role.kubernetes.io/master
+NAME                           STATUS   ROLES    AGE     VERSION
+ip-10-0-131-183.ec2.internal   Ready    master   6h13m   v1.17.1
+ip-10-0-164-97.ec2.internal    Ready    master   6h13m   v1.17.1
+ip-10-0-154-204.ec2.internal   Ready    master   6h13m   v1.17.1
+----
+
+.. Check whether the status of an etcd Pod is either `Error` or `CrashloopBackoff`:
++
+----
+$ oc get pods -n openshift-etcd | grep etcd
+etcd-ip-10-0-131-183.ec2.internal                2/3     Error       7          6h9m <1>
+etcd-ip-10-0-164-97.ec2.internal                 3/3     Running     0          6h6m
+etcd-ip-10-0-154-204.ec2.internal                3/3     Running     0          6h6m
+----
+<1> Since this status of this Pod is `Error`, then the *node is running, but etcd is crashlooping*.
+
+// TODO: Are there any other possible situations other than the above 2 that an unhealthy member could be in? If so, which procedure should be the default?
```

Comment:
@alaypatel07 PTAL at this question

Reply:
actually there is one:

If the admin knows that the underlying node is NotReady or machine is not Running and the admin expects the same machine/node to be `Ready/Running` soon, they do not need to perform any steps, cluster etcd operator will automatically sync when the unhealthy node/machine goes into expected state

Reply:
Thanks, I can work in a mention of this.

Reply:
@alaypatel07 I added a note here on this: https://github.com/openshift/openshift-docs/pull/20890/files#diff-1fca3c59dde22eaa3ca910449ad421d0R17

Can you see if the wording is okay?